### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by ergz**
this PR adds in some refactor to the way api status codes are responded to from the API.

I have a question about the report and how it is assigned to a variable in the global env, comment added in that code line below.

### CODE REVIEW

File: R/urls.R [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#' @title Base Urls
+BASE_URLS <- c(
+"staging" = "https://pasta-s.lternet.edu",
+"development" = "https://pasta-d.lternet.edu",
+"production" = "https://pasta.lternet.edu"
```
Comment (InigoP) [inline R/urls.R:5]: url missing "/" in the end 
```suggestion
"staging" = "https://pasta-s.lternet.edu/",
"development" = "https://pasta-d.lternet.edu/",
"production" = "https://pasta.lternet.edu/"
```

File: R/urls.R [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#' @title Base Urls
+BASE_URLS <- c(
+"staging" = "https://pasta-s.lternet.edu",
```
Comment (InigoP) [inline R/urls.R]:
```suggestion
"staging" = "https://pasta-s.lternet.edu/",
```

File: R/urls.R [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+"development" = "https://pasta-d.lternet.edu",
```
Comment (InigoP) [inline R/urls.R]:
```suggestion
"development" = "https://pasta-d.lternet.edu/",
```

File: R/urls.R (hunk 1 above)
Comment (InigoP) [inline R/urls.R]:
```suggestion
"production" = "https://pasta.lternet.edu/"
```

**Comment by alvizek** [comment]
Inigo and I paired to test these changes:
1. tried uploaded jpe-feather-edi package to staging as 1239.2 (1239.1 already existed on the EDI portal). Returned error - "{detail: not found}". After a couple more attempts and no success we checked the portal to find that 1239.2 had actually uploaded.
2. We then tried using the main branch V0.1 to upload the package as 1239.3. This took a while but uploaded successfully with the expected message.
3. Guessing that the new error handling code is categorizing as a success as an error and looking for error message when there is none.
4. Added print response code. When POST - 202, when GET (to look for error) - 404. Assuming that the `400L` isn't working right to match with code of 404 so going to the `else` statement. This follows with pasta+ docs (below).

GET : /error/eml/{transaction}
404 Not Found	No error associated with the specified data package is found

- There was also a wrong url and learned that it will continue polling until it times out

5. After Inigo updated the url and the 400L -> 404L worked

File: R/api-functions.R [hunk 4]
```diff
@@ -209,72 +207,76 @@ upload_edi_package <- function(user_id, password, eml_file_path, environment = "
...
-        }
-        # Stop loop if iterating through more than 5 times 
-        else if(iter > max_iter) {
-          stop("Request timed out, check that you inputs are all valid, rerun evalutate_edi_package(), and try again")
-        }
+      cli::cli_abort(c(
+        "during the upload process", 
+        "x" = "api responded with the following: {error_message}"
+      ))
+    } else if (identical(check_error$status, 400L)) { # no error found
```
Comment (InigoP) [inline R/api-functions.R]:
```suggestion
    } else if (identical(check_error$status, 404L)) { # no error found 
```

File: R/api-functions.R [hunk 5]
```diff
@@ -209,72 +207,76 @@ upload_edi_package <- function(user_id, password, eml_file_path, environment = "
...
-        else if(iter > max_iter) {
-          stop("Request timed out, check that you inputs are all valid, rerun evalutate_edi_package(), and try again")
-        }
+      cli::cli_abort(c(
+        "during the upload process", 
+        "x" = "api responded with the following: {error_message}"
+      ))
+    } else if (identical(check_error$status, 400L)) { # no error found 
+      poll_response <- poll_endpoint_at_dynamic_interval(
+        endpoint = httr::modify_url(base_url, path = glue::glue("package/report/eml{scope}/{identifier}/{revision}")),
```
Comment (InigoP) [inline R/api-functions.R]:
```suggestion
        endpoint = httr::modify_url(base_url, path = glue::glue("package/report/eml/{scope}/{identifier}/{revision}")), 
```

File: R/api-functions.R [hunk 6]
```diff
@@ -209,72 +207,76 @@ upload_edi_package <- function(user_id, password, eml_file_path, environment = "
...
-        }
+      cli::cli_abort(c(
+        "during the upload process", 
+        "x" = "api responded with the following: {error_message}"
+      ))
+    } else if (identical(check_error$status, 400L)) { # no error found 
+      poll_response <- poll_endpoint_at_dynamic_interval(
+        endpoint = httr::modify_url(base_url, path = glue::glue("package/report/eml{scope}/{identifier}/{revision}")), 
+        user_id = user_id, 
+        password = pasword,
```
Comment (InigoP) [inline R/api-functions.R]:
```suggestion
        password = password,
```

File: R/api-functions.R [hunk 7]
```diff
@@ -209,72 +207,76 @@ upload_edi_package <- function(user_id, password, eml_file_path, environment = "
...
+        time_out_seconds = .max_timeout * 60, 
+        verbose = TRUE
+      )
+      
+      if (identical(poll_response$status_code, 200L)) {
+        cli::cli_alert_success("data package poasted to EDI. Check EDI {environment} to confirm")
```
Comment (InigoP) [inline R/api-functions.R]:
```suggestion
        cli::cli_alert_success("data package posted to EDI. Check EDI {environment} to confirm")
```

File: R/api-functions.R [hunk 8]
```diff
@@ -209,72 +207,76 @@ upload_edi_package <- function(user_id, password, eml_file_path, environment = "
...
+        return()
```
Comment (InigoP) [inline R/api-functions.R]: Currently returning NULL - update with return status?

**Review by alvizek — APPROVED** [review]
Inigo and I reviewed the changes Emanuel made and realized that one of the suggested changes was not captured. We implemented this change and then tested the updates on the Feather River package on staging. The update worked. Approving changes

**Pull request merged**